### ISSUE TITLE
Duplicate assets when referencing a chain of nuget packages

### ISSUE THREAD

**Issue by jeromelaban** [microsoft/microsoft-ui-xaml#8857]
### Describe the bug

Given an app that references a nuget package called `PackageA`, which itself references a nuget package called `PackageB`. 

If `PackageB` contains an asset called `File01.txt`, the creation of the `PackageA` package will also include a reference to `File01.txt` in its own `pri` file.

When the app references `PackageA`, assets from `PackageA` and `PackageB` are elected to be placed in the app layout, causing the file name `File01.txt` to be present twice, and will cause the build to fail.

### Steps to reproduce the bug

- Create a nuget package `MyPackage01` which references `Uno.Material.WinUI`
- Take the nuget package and place it in a local folder source for nuget
- In a separate solution, create an app that references the `MyPackage01` nuget package (not a direct project reference)
- Build the app and you'll notice that there duplicate layout files.

### Expected behavior

App build works properly when using these packages.

### NuGet package version

WinUI 3 - Windows App SDK 1.4.0: 1.4.230822000

### Additional context

In most of Uno Platform packages, we're now including this workaround (works with both UWP and WinUI) :

```xml
<!-- Workaround to avoid duplicate assets https://github.com/microsoft/microsoft-ui-xaml/issues/8857 -->
<Target Name="AdjustGetPackagingOutput1" AfterTargets="GetMrtPackagingOutputs">
	<Message Importance="high" Text="Applying NuGet packaging workaround for dependent PRI files exclusion" />
	<ItemGroup>
		<_OtherPriFiles Include="@(PackagingOutputs)" Condition="'%(Extension)' == '.pri' and ('%(PackagingOutputs.ReferenceSourceTarget)' == 'ProjectReference' or '%(PackagingOutputs.NugetSourceType)'=='Package')" />
		<PackagingOutputs Remove="@(_OtherPriFiles)" />
	</ItemGroup>
</Target>
<Target Name="AdjustGetPackagingOutput2" BeforeTargets="AddPriPayloadFilesToCopyToOutputDirectoryItems">
	<Message Importance="high" Text="Applying NuGet packaging workaround for dependent PRI files exclusion" />
	<ItemGroup>
		<_OtherPriFiles1 Include="@(_ReferenceRelatedPaths)" Condition="'%(Extension)' == '.pri' and ('%(_ReferenceRelatedPaths.ReferenceSourceTarget)' == 'ProjectReference' or '%(_ReferenceRelatedPaths.NugetSourceType)'=='Package')" />
		<_ReferenceRelatedPaths Remove="@(_OtherPriFiles1)" />
		<_OtherPriFiles2 Include="@(ReferenceCopyLocalPaths)" Condition="'%(Extension)' == '.pri' and ('%(ReferenceCopyLocalPaths.ReferenceSourceTarget)' == 'ProjectReference' or '%(ReferenceCopyLocalPaths.NugetSourceType)'=='Package')" />
		<ReferenceCopyLocalPaths Remove="@(_OtherPriFiles2)" />
	</ItemGroup>
</Target>
```

which prevents files from direct nuget package references to be incorrectly included in a package being built.

**Comment by duncanmacmichael** [comment]
Fixed in the 1.4.5 release; marking as completed.

**Comment by SongOfYouth** [comment]
> Fixed in the 1.4.5 release; marking as completed.

The xbf file of nuget package lost when i build the exe project which referenced the package, then raise 'xaml parse error' after run it.
